cmd/hunt.go

```diff
@@ -38,23 +38,22 @@ var huntCmd = &cobra.Command{
 	Long: `
 `,
 	Example: "  harpoon hunt --file .harpoon.yaml",
-	Run: func(cmd *cobra.Command, args []string) {
+	RunE: func(cmd *cobra.Command, args []string) error {
 		file, err := os.Open(harpoonFile)
 		if err != nil {
-			fmt.Printf("failed to open %s: %v\n", harpoonFile, err)
-			return
+			return fmt.Errorf("failed to open %s: %w", harpoonFile, err)
 		}
 		defer file.Close()
 
 		byteValue, err := io.ReadAll(file)
 		if err != nil {
-			fmt.Printf("Failed to read file: %s", err)
+			return fmt.Errorf("failed to read file: %w", err)
```

Comment:
Adding the file name in the error could help,
you did it for other errors

```suggestion
			return fmt.Errorf("failed to read filev %q: %w", harpoonFile, err)
```

Maybe you could use %q everywhere, BTW.

It's a %s surrounded with quotes

Reply:
there's a typo in `filev %q`